drivers/crypto/crypto_mchp_xec_symcr.c

```diff
@@ -0,0 +1,571 @@
+
+	ret = mchp_xec_rom_hash_feed_wrapper(c, (const uint8_t *)pkt->in_buf, fill_len);
+	if (ret) {
+		LOG_ERR("ROM hash feed error %d", ret);
+		return ret;
+	}
+
+	if (finish) {
+		ret = mchp_xec_rom_hash_digest_wrapper(c, pkt->out_buf);
+		if (ret) {
+			LOG_ERR("ROM Hash final error %d", ret);
+			return ret;
+		}
+	} else {
+		ret = mchp_xec_rom_hash_save_state(c);
+		if (ret) {
+			LOG_ERR("ROM hash save state error %d", ret);
+			return ret;
+		}
+	}
+	ret = mchp_xec_rom_hash_wait_wrapper(c);
+	if (ret) {
+		LOG_ERR("ROM hash wait error %d", ret);
+		return ret;
+	}
+	if (finish) {
+		hs->blklen = 0;
+	} else {
+		memcpy(hs->blockbuf, &pkt->in_buf[fill_len], rem_len);
+		hs->blklen = rem_len;
+	}
+
+	return 0;
+}
+
+static int xec_symcr_hash_session_begin(const struct device *dev, struct hash_ctx *ctx,
+					enum hash_algo algo)
+{
+	struct xec_symcr_data *data = dev->data;
+	struct xec_symcr_hash_session *hs = NULL;
+	struct mchphash *c = NULL;
+	struct mchphashstate *cstate = NULL;
+	enum mchp_rom_hash_alg_id rom_algo = MCHP_ROM_HASH_ALG_NONE;
+	int session_idx = 0;
+	int ret = 0;
+
+	if (!ctx) {
+		LOG_ERR("Bad parameters");
+		return -EINVAL;
+	}
+
+	if (ctx->flags & ~(MCHP_XEC_SYMCR_CAPS_SUPPORT)) {
+		LOG_ERR("Unsupported flag");
+		return -EINVAL;
+	}
+
+	rom_algo = lookup_hash_alg(algo);
+	if (rom_algo == MCHP_ROM_HASH_ALG_NONE) {
+		LOG_ERR("Unsupported algo %d", algo);
+		return -EINVAL;
+	}
+
+	session_idx = mchp_xec_get_unused_session_index(data);
+	if (session_idx < 0) {
+		LOG_ERR("No session available");
+		return -ENOSPC;
+	}
+
+	hs = &data->hash_sessions[session_idx];
+
+	hs->algo = algo;
+	hs->rom_algo = rom_algo;
+	hs->open = false;
+	hs->blklen = 0;
+	hs->blksz = hash_block_size(algo);
+
+	ctx->drv_sessn_state = hs;
+	ctx->started = false;
+	ctx->hash_hndlr = xec_symcr_do_hash;
+
+	/* reset HW at beginning of session */
+	ret = mchp_xec_rom_ah_dma_init(MCHP_ROM_AH_DMA_INIT_WITH_RESET);
+	if (ret) {
+		LOG_ERR("ROM HW init error %d", ret);
+		return -EIO;
+	}
+
+	c = &hs->mhctx;
+	cstate = &hs->mhstate;
+
+	ret = init_rom_hash_context(hs->rom_algo, c);
+	if (ret) {
+		LOG_ERR("ROM HW context init error %d", ret);
+		return ret;
+	}
+
+	ret = mchp_xec_rom_hash_init_state_wrapper(c, cstate, hs->statebuf);
+	if (ret) {
+		LOG_ERR("ROM HW init state error %d", ret);
+	}
+
+	hs->open = true;
+
+	return ret;
+}
+
+/*
+ * struct hash_ctx {
+ *	const struct device *device; this device driver's instance structure
+ *	void *drv_sessn_state; pointer to driver instance struct session state. Defined by driver
+ *	hash_op hash_hndlr; pointer to this driver function. App calls via pointer to do operations
+ *	bool started; true if multipart hash has been started
+ *	uint16_t flags; app populates this before calling hash_begin_session
+ * }
+ */
+static int xec_symcr_hash_session_free(const struct device *dev, struct hash_ctx *ctx)
+{
+	struct xec_symcr_hash_session *hs = NULL;
+	int ret = 0;
+
+	if (!dev || !ctx) {
+		return -EINVAL;
+	}
```

Comment:
This check is not needed here. The API has already referenced these parameters before calling it.

Reply:
Logic removed to check against !NULL.